src/cli.js

```diff
@@ -681,6 +702,55 @@ var exports = {
   },
 
   /**
+   * Helper to get proper reporter.
+   *
+   * @param {string} name of reporter to fetch, pre-defined or custom.
+   */
+  getReporter: function(reporterName) {
+    var reporter = defReporter; // default
+
+    if (!reporterName) {
+      return reporter; // return default
+    }
+
+    switch (reporterName) {
+      // JSLint reporter
+      case "jslint":
+        reporter = "./reporters/jslint_xml.js";
+        break;
+
+      // CheckStyle (XML) reporter
+      case "checkstyle":
+        reporter = "./reporters/checkstyle.js";
+        break;
+
+      // Unix reporter
+      case "unix":
+        reporter = "./reporters/unix.js";
+        break;
+
+      // Reporter that displays additional JSHint data
+      case "non_error":
+        reporter = "./reporters/non_error.js";
+        break;
+
+      // Custom reporter. not to be confused with 'default' if undefined above
+      default:
+        reporter = path.resolve(process.cwd(), reporterName);
+        break;
+    }
+
+    if (reporter) {
```

Comment:
Agreed. The code handles false/null/"". This 'default:' was an interpretation of the old switch statement that evaluated (true) and then had this block for detecting custom reporters.

The 'default' in this new switch statement is for handling custom reporters.

Am I understanding the requirements correctly?

##Original 'default' line
```javascript
// Custom reporter
    case options.reporter !== undefined:
      options.reporter = path.resolve(process.cwd(), options.reporter);
```